server/src/main/java/org/elasticsearch/indices/recovery/RecoveryState.java

```diff
@@ -142,12 +145,17 @@ public RecoveryState(StreamInput in) throws IOException {
         translog = new Translog(in);
         verifyIndex = new VerifyIndex(in);
         primary = in.readBoolean();
+        if (in.getVersion().after(Version.V_7_12_0)) {
```

Comment:
I've used this version guard temporarily since I'm not really sure how to backport this PR without breaking BWC tests.